packages/coin-evm/src/index.ts

```diff
@@ -93,7 +94,11 @@ class Evm extends COIN.ECDSACoin {
       );
     }
 
-    const script = SCRIPTS.signSmartContractTransaction.scriptWithSignature;
+    let script = SCRIPTS.signSmartContractTransaction.scriptWithSignature;
+    // Staking contract for Fantom
+    if(this.chain.id === FANTOM.id && client.transaction.to.toLowerCase() === FANTOM.stakingContract.toLowerCase()){
```

Comment:
The staking contract might apply different methods. Should the condition be more precise for corresponding signing scripts ?

Reply:
You mean checking data for program id, and allowing only certain known program, for example: delegation (0x9fa6dd35) and withdraw (0x0962ef79)?

Reply:
Yes, we are used to handle these staking methods by showing method names on screen. You could refer to the procedure of ETH smart contracts in the SDK, or you could ask PO for proper implementation in the task.

Reply:
Staking detection is more precise and the display is updated according to PO request.